edb/server/pgconnparams.py

```diff
@@ -66,195 +62,23 @@ class ConnectionParameters:
     import ctypes.wintypes
 
     CSIDL_APPDATA = 0x001a
-    PGPASSFILE = 'pgpass.conf'
 
-    def get_pg_home_directory() -> Optional[pathlib.Path]:
+    def get_pg_home_directory() -> pathlib.Path:
         # We cannot simply use expanduser() as that returns the user's
         # home directory, whereas Postgres stores its config in
         # %AppData% on Windows.
         buf = ctypes.create_unicode_buffer(ctypes.wintypes.MAX_PATH)
         r = ctypes.windll.shell32.SHGetFolderPathW(  # type: ignore
             0, CSIDL_APPDATA, 0, 0, buf)
         if r:
-            return None
+            # Fall back to the home directory
+            warnings.warn("Could not resolve %AppData%", stacklevel=2)
+            return pathlib.Path.home()
```

Comment:
Good point. I switched this to raise an exception -- it is an extremely unlikely code path, and only on Windows in the case where the OS has spectacularly failed.

Reply:
Due to a git failure, this last commit failed to push. I'll return to this code in the future as it will likely become pure Rust and we can revisit them.